### PR TITLE
Enemy Auto Path Finding and Other Refactoring

### PR DESCRIPTION
**Pull request by DKevi**
Enemy Added.

### CODE REVIEW

File: Assets/Script/Enemy/EnemyStat.cs
Comment (AlstonXiao) [inline Assets/Script/Enemy/EnemyStat.cs:1]: Should this file be removed from this PR?

File: Assets/Script/CharacterStats.cs
Comment (AlstonXiao) [inline Assets/Script/CharacterStats.cs:1]: Will this file be contradicted with our current backend design?

File: Assets/Script/MovementManager.cs
Comment (AlstonXiao) [inline Assets/Script/MovementManager.cs:1]: Empty file?

**Review by AlstonXiao — COMMENTED** [review]
需要一点clean up，你可以把你现在的folder git branch出去，然后把这个新branch当成你的working branch。你现在pr的这个branch就留下来做点cleanup。等pr通过了你换到你的新branch上可以继续follow你的教程

**Comment by AlstonXiao** [comment]
LGTM, 但是character stat能不能改名成enemy stat因为character stat会和backend的character stat冲突

**Review by zyd2001 — APPROVED** [review]
NICE

**Pull request merged**